crates/bevy_transform/src/components/global_transform.rs

```diff
@@ -74,11 +74,36 @@ impl GlobalTransform {
         Mat4::from_scale_rotation_translation(self.scale, self.rotation, self.translation)
     }
 
+    #[inline]
+    pub fn right(&self) -> Vec3 {
+        self.rotation * Vec3::unit_x()
```

Comment:
If Bevy is right-handed y-up, and we're calling +z forward, then doesn't that make +x _left_, not right?

(Sorry about my previous deleted comment; got momentarily confused about the whole "-handed" naming convention.)

Reply:
I.e. I expect the direction called "left" to be to my left if I'm looking down the direction called "forward".

Reply:
Ah! Unity (referenced above) is left-handed — maybe that's where the oddness came from. I was beginning to worry that there were two different conventions for what's called "left", i.e. because -x is left _on screen_ in Bevy when doing 2D rendering. :sweat_smile:

Reply:
haha yeah fortunately left still means left :smile:

Reply:
I think I communicated that poorly. I was actually trying to say that I think the code in this PR is actually "wrong" as it stands, at least in that by one obvious interpretation it appears internally inconsistent.

Let me try again. :blush: 

Unity, which appears to have inspired this code, is left-handed. In that left-handed context, all of the function names in this PR would actually match up to their implementations.

But Bevy is right-handed, and so they appear to be inconsistent. If `forward` means "along the positive z-axis", as it is defined in this PR, then `right` should be whatever direction you end up facing by turning 90 degrees clockwise from forward. Which in Bevy would mean you are now pointing along the _negative_ x-axis, whereas this PR defines `right` as pointing down the _positive_ x-axis.

I _suspect_ the confusion exists because of mixing the idea of absolute directions in screen space with relative orientation in 3D space — and in particular the fact that you can actually do that without consequence in Unity, but _not_ in Bevy. Or of course it could be as simple as just having copied the definitions from Unity without correcting for the differing handedness.

_Or_... there could be a third explanation that I haven't thought of, which makes the code in this PR "as intended". But if that's so, then I feel this whole API as it stands — at least in the context of a right-handed coordinate system like Bevy's — is likely to provide more confusion than utility, because there are so many obvious interpretations of it that make it _look_ wrong.

I hope I'm making sense... :sweat_smile:

Reply:
Ahh yeah I see your point now l. And I do think you were clear enough in your initial description, I just didn't think enough about it / fixated too much on your last message. In 2d this "works" because +x is "the viewer's right". However from the perspective of the transform, +x is left (which is also true for 3d).

This should definitely be fixed. I think the functionality is still useful, but we need new names for it. Maybe just something like "local_positive_x" and "local_negative_x"?

We could also consider just flipping the signs on the x-axis, but that also makes things a little weird in 2d, as "left" would be the viewer's "right".

If anyone has better ideas, let me know :)

Reply:
I needed these functions when making a 2d game, and I tested it in that context, so I didn't choose the +x direction because of Unity but because it seemed to do the right thing.